### ISSUE TITLE
[2.5] [Backport] [Rancher-webhook] Add support for hardened cluster

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#34320]
backport https://github.com/rancher/rancher/issues/34188

**Comment by jiaqiluo** [comment]
available in the chart `rancher-webhook 0.1.200-rc1+up0.1.2-rc1`

**Comment by brendarearden** [comment]
available to test in v2.5.10-rc1

**Comment by anupama2501** [comment]
Verified on rancher v2.5.10-rc1

- created an rke cluster by following the doc to harden the cluster 
- https://rancher.com/docs/rancher/v2.5/en/security/rancher-2.5/1.6-hardening-2.5/#reference-hardened-rke-cluster-yml-configuration
- Once the cluster is up, installed rancher on this rke cluster via helm:

```
helm install rancher rancher-latest/rancher \
  --namespace cattle-system \
  --version ${RANCHER_VERSION} \
  --set hostname=${HOSTNAME} --create-namespace
```

- Verified the rancher server HA has come up fine and no errors were seen during rancher installation.
- Webhook workload in the system project has come up fine with no errors.

```
kubectl get MutatingWebhookConfiguration
NAME                   WEBHOOKS   AGE
cert-manager-webhook   1          16m
rancher.cattle.io      1          4m40s
```
